rust/src/transformations/make_stable_expr/expr_cast/mod.rs

```diff
@@ -0,0 +1,78 @@
+use polars::chunked_array::cast::CastOptions;
+use polars::prelude::*;
+use polars_plan::dsl::Expr;
+
+use crate::core::{Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, OuterMetric, SeriesDomain};
+use crate::error::*;
+use crate::polars::ExprFunction;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that returns a `cast(dtype)` expression for a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The clipping expression
+pub fn make_expr_cast<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Cast {
+        expr: input,
+        data_type: to_type,
+        mut options,
+    } = expr
+    else {
+        return fallible!(MakeTransformation, "expected function expression");
+    };
+
+    // strict casting makes the transformation unstable: errors tell you things about private data
+    // Could throw an error if strict, but it is the default,
+    // so for ease-of-use it is forced to be non-strict
+    if let CastOptions::Strict = options {
```

Comment:
I don't think I understand the syntax here: Looks like it's assigning to `CastOptions::Strict`?

Reply:
`if let` is usually used to bind to a variable, but I didn't bind a variable there, I just used it to match an enum variant. The new code makes this more clear.